File: sdks/kotlin/src/main/kotlin/com/opencat/sdk/Configuration.kt
package com.opencat.sdk

import android.content.Context

sealed class Configuration {
    abstract val context: Context
    abstract val appUserId: String

    data class StandaloneConfiguration(
        override val context: Context,
        override val appUserId: String,
    ) : Configuration()

    data class ServerConfiguration(
        override val context: Context,
        val serverUrl: String,
        val apiKey: String,
        override val appUserId: String,
    ) : Configuration()
}

File: sdks/flutter/lib/src/configuration.dart
/// Configuration for standalone mode (no server, on-device only).
class StandaloneConfiguration {
  final String appUserId;

  const StandaloneConfiguration({required this.appUserId});
}

/// Configuration for server mode (full features with OpenCat backend).
class ServerConfiguration {
  final String serverUrl;
  final String apiKey;
  final String appUserId;

  const ServerConfiguration({
    required this.serverUrl,
    required this.apiKey,
    required this.appUserId,
  });
}
